src/openms/source/QC/DBSuitability.cpp

```diff
@@ -101,102 +91,51 @@ namespace OpenMS
-      }
-
-      // find the second target hit, skip all decoy or novo hits inbetween
-      const PeptideHit* second_hit = nullptr;
-      for (UInt i = 1; i < hits.size(); ++i)
-      {
-        // check for FDR
-        if (!passesFDR_(hits[i], FDR)) break;
-
-        // check if target, also check for "target+decoy" value
-        String td_info(hits[i].getMetaValue("target_decoy"));
-        if (td_info.find("target") != 0) continue;
-
-        // check if hit is novo hit
-        if (isNovoHit_(hits[i])) continue;
-
-        second_hit = &hits[i];
-        break;
-      }
-      if (second_hit == nullptr) // no second target hit with given FDR found
-      {
-        ++data.num_top_novo;
-        continue;
-      }
-
-      // second hit is db hit
-      ++data.num_interest;
-
-      // check for re-ranking
-      if (no_rerank)
-      {
-        ++data.num_top_novo;
-        continue;
-      }
-
-      // check for xcorr score
-      if (!top_hit.metaValueExists("MS:1002252") || !second_hit->metaValueExists("MS:1002252"))
-      {
-        throw(Exception::MissingInformation(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "No cross correlation score found at peptide hit. Only Comet search engine is supported right now."));
-      }
-
-      double top_xscore_mw = double(top_hit.getMetaValue("MS:1002252")) / top_hit.getSequence().getMonoWeight();
-      double second_xscore_mw = double(second_hit->getMetaValue("MS:1002252")) / second_hit->getSequence().getMonoWeight();
-      if (top_xscore_mw - second_xscore_mw <= data.cut_off)
-      {
-        ++data.num_top_db;
-        ++data.num_re_ranked;
-      }
-      else
-      {
-        ++data.num_top_novo;
-      }
-    }
-
-    if (data.num_top_db == 0 && data.num_top_novo == 0)
-    {
-      OPENMS_LOG_WARN << "Identifications could not be assigned to either the database or the deNovo protein. Probably your FDR threshold is too strict." << endl;
-      data.suitability = DBL_MAX;
-      return;
-    }
+    // calculate suitability
+    SuitabilityData d;
+    results_.push_back(d);
+    SuitabilityData& data = results_.back();
+    calculateSuitability_(pep_ids, data);
 
-    data.suitability = double(data.num_top_db) / (data.num_top_db + data.num_top_novo);
+    // calculate correction of suitability with extrapolation
+    
+    // sampled run
+    // maybe multiple runs? could be controlled with a parameter
+    double ratio = 0.5;
+    vector<FASTAFile::FASTAEntry> sampled_db = getSubsampledFasta_(original_fasta, ratio);
+    sampled_db.insert(sampled_db.end(), novo_fasta.begin(), novo_fasta.end());
+    calculateDecoys_(sampled_db);
+    vector<PeptideIdentification> subsampled_ids = runIdentificationSearch_(exp, sampled_db, search_info.first, search_info.second);
+
+    SuitabilityData sampled_data;
+    calculateSuitability_(subsampled_ids, sampled_data);
+
+    // slopes of db and deNovo hits
+    double db_slope = (int(sampled_data.num_top_db) - int(data.num_top_db)) / (-ratio);
```

Comment:
`*data` is very generic.
Prepend with `suitability_` + `_sampled/_full` instead?

Reply:
technically, you need `full - sampled / (1 - ratio)`, right?